engine/src/main/java/org/destinationsol/game/console/commands/RespawnCommandHandler.java

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright 2019 MovingBlocks
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.destinationsol.game.console.commands;
+
+import org.destinationsol.game.Console;
+import org.destinationsol.game.Hero;
+import org.destinationsol.game.SolGame;
+import org.destinationsol.game.console.ConsoleInputHandler;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * A command used to respawn player's ship
+ */
+public class RespawnCommandHandler implements ConsoleInputHandler {
+
+    private Hero hero;
+    private SolGame game;
+
+    public RespawnCommandHandler(Hero hero, SolGame game) {
+        this.hero = hero;
+        this.game = game;
+    }
+
+    private static Logger logger = LoggerFactory.getLogger(RespawnCommandHandler.class);
+
+    @Override
+    public void handle(String input, Console console) {
+        if (hero.isAlive()) {
+            logger.warn("Cannot respawn hero when not dead!");
+            console.println("Cannot respawn hero when not dead!");
```

Comment:
I'd much rather have `console.println()` handle logging so that we don't need to do this every time.

Reply:
Yeah, would be better, I could re-work this in another PR, as this is a bit more unrelated and I have few other ideas that would add to this.